### ISSUE TITLE
Sse2.Compare*** tests fail on Windows arm

### ISSUE THREAD

**Issue by echesakovMSFT** [dotnet/coreclr#19853]
All Sse2.Compare-tests fail on **Windows_NT arm Cross Checked Innerloop Build and Test** (for example, see https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm_cross_checked_windows_nt_innerloop_tst_prtest/1534/)

@CarolEidt @tannergooding @4creators 

Please find below the list of failed tests:
```
Tests.lst=CompareLessThanUnorderedScalar_r.cmd_11409, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanUnorderedScalar_r\CompareLessThanUnorderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrEqualOrderedScalar_r.cmd_11426, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrEqualOrderedScalar_r\CompareLessThanOrEqualOrderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotEqualScalar_r.cmd_11435, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotEqualScalar_r\CompareNotEqualScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotEqualScalar_ro.cmd_11442, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotEqualScalar_ro\CompareNotEqualScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareOrderedScalar_r.cmd_11446, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareOrderedScalar_r\CompareOrderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareEqualUnorderedScalar_r.cmd_11456, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareEqualUnorderedScalar_r\CompareEqualUnorderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrderedScalar_r.cmd_11457, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrderedScalar_r\CompareGreaterThanOrderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanScalar_r.cmd_11460, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanScalar_r\CompareGreaterThanScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrderedScalar_ro.cmd_11467, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrderedScalar_ro\CompareGreaterThanOrderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotGreaterThanOrEqualScalar_r.cmd_11482, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotGreaterThanOrEqualScalar_r\CompareNotGreaterThanOrEqualScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanScalar_r.cmd_11487, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanScalar_r\CompareLessThanScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanScalar_ro.cmd_11511, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanScalar_ro\CompareGreaterThanScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrEqualUnorderedScalar_r.cmd_11528, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrEqualUnorderedScalar_r\CompareGreaterThanOrEqualUnorderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotLessThanOrEqualScalar_r.cmd_11529, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotLessThanOrEqualScalar_r\CompareNotLessThanOrEqualScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareEqualOrderedScalar_r.cmd_11530, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareEqualOrderedScalar_r\CompareEqualOrderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrEqualOrderedScalar_ro.cmd_11564, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrEqualOrderedScalar_ro\CompareLessThanOrEqualOrderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotLessThanScalar_r.cmd_11582, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotLessThanScalar_r\CompareNotLessThanScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrEqualOrderedScalar_ro.cmd_11584, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrEqualOrderedScalar_ro\CompareGreaterThanOrEqualOrderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotGreaterThanScalar_ro.cmd_11585, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotGreaterThanScalar_ro\CompareNotGreaterThanScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrEqualOrderedScalar_r.cmd_11586, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrEqualOrderedScalar_r\CompareGreaterThanOrEqualOrderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotLessThanScalar_ro.cmd_11604, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotLessThanScalar_ro\CompareNotLessThanScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrderedScalar_r.cmd_11619, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrderedScalar_r\CompareLessThanOrderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrEqualScalar_r.cmd_11624, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrEqualScalar_r\CompareGreaterThanOrEqualScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrEqualUnorderedScalar_r.cmd_11636, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrEqualUnorderedScalar_r\CompareLessThanOrEqualUnorderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrEqualScalar_r.cmd_11637, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrEqualScalar_r\CompareLessThanOrEqualScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareOrderedScalar_ro.cmd_11653, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareOrderedScalar_ro\CompareOrderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrEqualScalar_ro.cmd_11663, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrEqualScalar_ro\CompareGreaterThanOrEqualScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotEqualOrderedScalar_r.cmd_11676, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotEqualOrderedScalar_r\CompareNotEqualOrderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrEqualUnorderedScalar_ro.cmd_11685, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrEqualUnorderedScalar_ro\CompareLessThanOrEqualUnorderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareEqualUnorderedScalar_ro.cmd_11691, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareEqualUnorderedScalar_ro\CompareEqualUnorderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareEqualOrderedScalar_ro.cmd_11697, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareEqualOrderedScalar_ro\CompareEqualOrderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrderedScalar_ro.cmd_11703, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrderedScalar_ro\CompareLessThanOrderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotLessThanOrEqualScalar_ro.cmd_11742, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotLessThanOrEqualScalar_ro\CompareNotLessThanOrEqualScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareUnorderedScalar_r.cmd_11743, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareUnorderedScalar_r\CompareUnorderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareEqualScalar_ro.cmd_11748, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareEqualScalar_ro\CompareEqualScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareUnorderedScalar_ro.cmd_11750, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareUnorderedScalar_ro\CompareUnorderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanUnorderedScalar_ro.cmd_11752, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanUnorderedScalar_ro\CompareLessThanUnorderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanScalar_ro.cmd_11763, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanScalar_ro\CompareLessThanScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotGreaterThanOrEqualScalar_ro.cmd_11775, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotGreaterThanOrEqualScalar_ro\CompareNotGreaterThanOrEqualScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanUnorderedScalar_ro.cmd_11782, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanUnorderedScalar_ro\CompareGreaterThanUnorderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanUnorderedScalar_r.cmd_11787, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanUnorderedScalar_r\CompareGreaterThanUnorderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareEqualScalar_r.cmd_11802, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareEqualScalar_r\CompareEqualScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotGreaterThanScalar_r.cmd_11803, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotGreaterThanScalar_r\CompareNotGreaterThanScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotEqualOrderedScalar_ro.cmd_11832, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotEqualOrderedScalar_ro\CompareNotEqualOrderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareGreaterThanOrEqualUnorderedScalar_ro.cmd_11837, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareGreaterThanOrEqualUnorderedScalar_ro\CompareGreaterThanOrEqualUnorderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotEqualUnorderedScalar_r.cmd_11861, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotEqualUnorderedScalar_r\CompareNotEqualUnorderedScalar_r.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareNotEqualUnorderedScalar_ro.cmd_11870, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareNotEqualUnorderedScalar_ro\CompareNotEqualUnorderedScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
Tests.lst=CompareLessThanOrEqualScalar_ro.cmd_11879, , , Smrt00000001, , # JIT\HardwareIntrinsics\X86\Sse2\CompareLessThanOrEqualScalar_ro\CompareLessThanOrEqualScalar_ro.cmd  CATS: EXPECTED_PASS;NEW;EXCLUDED
```

**Comment by 4creators** [comment]
Thanks for reporting. Will investigate the problem look.

**Comment by 4creators** [comment]
From other PR #19677 which is rebased on top of Sse2.Compare tests CI jobs do not seem to run tests at all:

https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm64_cross_debug_windows_nt_innerloop_prtest/1140/consoleText

```cmd
BUILDTEST: Test build succeeded.  Finished at 16:17:36.32
BUILDTEST: Test binaries are available at D:\j\workspace\arm64_cross_d---ffc60a4b\bin\tests\Windows_NT.arm64.Debug
BUILD: Build succeeded.  Finished at 16:17:36.32
BUILD: Product binaries are available at D:\j\workspace\arm64_cross_d---ffc60a4b\bin\Product\Windows_NT.arm64.Debug
[arm64_cross_d---ffc60a4b] $ cmd /c call C:\Users\runner\AppData\Local\Temp\jenkins6465667221486837665.bat

D:\j\workspace\arm64_cross_d---ffc60a4b>powershell -NoProfile -Command "Add-Type -Assembly 'System.IO.Compression.FileSystem'; [System.IO.Compression.ZipFile]::CreateFromDirectory('.\bin\tests\Windows_NT.arm64.Debug', '.\bin\tests\tests.zip')" 

D:\j\workspace\arm64_cross_d---ffc60a4b>exit 0 
[Current build status] check if current [SUCCESS] is worse or equals then [SUCCESS] and better or equals then [SUCCESS]
Run condition [Current build status] enabling perform for step [[Archive the artifacts]]
Archiving artifacts
Setting status of e2f7c531de53b51c31881047aba76931885c4f97 to SUCCESS with url https://ci.dot.net/job/dotnet_coreclr/job/master/job/arm64_cross_debug_windows_nt_innerloop_prtest/1140/ and message: 'Build finished. '
Using context: Windows_NT arm64 Cross Debug Innerloop Build
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
Finished: SUCCESS
```

It's hard to know that anything can fail if it's not tested LOL
